trestle/core/commands/author/component.py

```diff
@@ -0,0 +1,290 @@
+            trestle_root = args.trestle_root
+            markdown_dir_name = args.output
+            profile_name = args.profile
+            component_name = args.name
+
+            return self.component_generate_all(
+                trestle_root, component_name, profile_name, markdown_dir_name, {}, None, False
+            )
+
+        except Exception as e:  # pragma: no cover
+            return handle_generic_command_exception(e, logger, 'Generation of the component markdown failed')
+
+    def component_generate_all(
+        self,
+        trestle_root: pathlib.Path,
+        comp_def_name: str,
+        profile_name: str,
+        markdown_dir_name: str,
+        yaml_header: Dict[str, Any],
+        sections_dict: Optional[Dict[str, str]],
+        overwrite_header_values: bool
+    ) -> int:
+        """Generate markdown for all components in comp def."""
+        if not file_utils.is_directory_name_allowed(markdown_dir_name):
+            raise TrestleError(f'{markdown_dir_name} is not an allowed directory name')
+        md_path = trestle_root / markdown_dir_name
+        md_path.mkdir(parents=True, exist_ok=True)
+        component_def, _ = load_validate_model_name(trestle_root, comp_def_name, comp.ComponentDefinition)
+        for component in as_list(component_def.components):
+            rc = self.component_generate_by_name(
+                trestle_root,
+                comp_def_name,
+                component.title,
+                profile_name,
+                md_path / component.title,
+                yaml_header,
+                sections_dict,
+                overwrite_header_values
+            )
+            if rc != CmdReturnCodes.SUCCESS.value:
+                return rc
+        return CmdReturnCodes.SUCCESS.value
+
+    def component_generate_by_name(
+        self,
+        trestle_root: pathlib.Path,
+        comp_def_name: str,
+        comp_name: str,
+        profile_name: str,
+        markdown_dir_path: pathlib.Path,
+        yaml_header: dict,
+        sections_dict: Optional[Dict[str, str]],
+        overwrite_header_values: bool
+    ) -> int:
+        """Create markdown based on the component and profile."""
+        profile_path = ModelUtils.full_path_for_top_level_model(trestle_root, profile_name, prof.Profile)
+        profile_resolver = ProfileResolver()
+        resolved_catalog = profile_resolver.get_resolved_profile_catalog(trestle_root, profile_path)
+        catalog_interface = CatalogInterface(resolved_catalog)
+        comp_def, _ = load_validate_model_name(trestle_root, comp_def_name, comp.ComponentDefinition)
+        context = ControlContext.generate(ContextPurpose.COMPONENT, True, trestle_root, markdown_dir_path)
+        context.sections_dict = sections_dict
+        context.prompt_responses = True
+        context.overwrite_header_values = overwrite_header_values
+        context.comp_def = comp_def
+        context.comp_name = comp_name
```

Comment:
This makes me think that having an initialiser with optional fields for `ControlContext` is better. Thus user can either create a default `ControlContext` (same as produced by `ControlContext.generate()`) but in the moments like this, when fields need to be customised, one can provide it in the initialisation `ControlContext(section_dict=..., promt_responses=...). 

I feel it keeps things more clean, what do you think?

Reply:
as discussed - leave for now but consider changes going forward.